### PR TITLE
add cancel and update offering, logging error for rfq.verifyClaims()

### PR DESCRIPTION
**Pull request by jiyoonie9**
## changes
* add cancel message
* update offering to include `cancellation` object
* logging error for rfq.VerifyClaims() as it was swallowing the error returned from internal calls

### CODE REVIEW

File: tbdex/cancel/cancel.go [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+	"github.com/tbd54566975/web5-go/dids/did"
+	"go.jetpack.io/typeid"
+)
+
+// Kind identifies this message kind
+const Kind = "cancel"
+
+// ValidNext returns the valid message kinds that can follow a cancel.
+func ValidNext() []string {
+	// todo hardcoded orderstatus kind here because otherwise i am introducing a circular dependency :(
```
Comment (jiyoonie9) [inline tbdex/cancel/cancel.go:21]: wondering if there's a way around this without hardcoding but could not find think of one myself

Comment (KendallWeihe) [reply tbdex/cancel/cancel.go:21]: is cancel not a terminal message? in which case nothing is valid after a cancel yeah?

Comment (jiyoonie9) [reply tbdex/cancel/cancel.go:21]: it is not - a cancel after an order indicates the customer wants a refund. if the exchange is subject to refund, then the rest of the exchange looks like PFI sending order statuses to Alice which says `here's the status of your refund` and finally closes the exchange with a Close

or if it's not refundable the PFI can just immediately send a Close which indicates `refund not allowed`

Comment (KendallWeihe) [reply tbdex/cancel/cancel.go:21]: ah right, okay thanks!

File: tbdex/offering/offering.go [hunk 2]
```diff
@@ -29,6 +29,13 @@ type Data struct {
 	Payin          *PayinDetails                 `json:"payin,omitempty"`
 	Payout         *PayoutDetails                `json:"payout,omitempty"`
 	RequiredClaims *pexv2.PresentationDefinition `json:"requiredClaims,omitempty"`
+	Cancellation   *Cancellation                 `json:"cancellation,omitempty"`
```
Comment (KendallWeihe) [inline tbdex/offering/offering.go:32]: not sure `omitempty` does anything meaningful here but from the spec this is a required data property

Comment (jiyoonie9) [reply tbdex/offering/offering.go:32]: i think omitempty is used to exclude the field when serializing to json

**Review by mistermoe — APPROVED** [review]

**Pull request merged**